spec/javascripts/govuk_publishing_components/load-analytics.spec.js

```diff
@@ -0,0 +1,323 @@
+  'register-trailer.service.gov.uk',
+  'register.ozone-depleting-substances.service.gov.uk',
+  'registertovote.service.gov.uk',
+  'register-vehicle.service.gov.uk',
+  'registers.service.gov.uk',
+  'reminders.mot-testing.service.gov.uk',
+  'renewable-heat-calculator.service.gov.uk',
+  'reply-jury-summons.service.gov.uk',
+  'report-director-conduct.service.gov.uk',
+  'report.fluorinated-gas.service.gov.uk',
+  'report.ozone-depleting-substances.service.gov.uk',
+  'right-to-rent.homeoffice.gov.uk',
+  'right-to-work.service.gov.uk',
+  'ruralpayments.service.gov.uk',
+  'schools-financial-benchmarking.service.gov.uk',
+  'secured.studentfinanceni.co.uk',
+  'secured.studentfinancewales.co.uk',
+  'selfservice.payments.service.gov.uk',
+  'send-money-to-prisoner.service.gov.uk',
+  'signin.service.gov.uk',
+  'sorn.service.gov.uk',
+  'staff.helpwithcourtfees.service.gov.uk',
+  'student-finance.service.gov.uk',
+  'tax.service.gov.uk',
+  'teacherservices.education.gov.uk',
+  'teaching-vacancies.service.gov.uk',
+  'to-visit-or-stay-in-the-uk.homeoffice.gov.uk',
+  'trade-tariff.service.gov.uk',
+  'tribunal-response.employmenttribunals.service.gov.uk',
+  'ukri.org',
+  'update-student-loan-employment-details.service.gov.uk',
+  'vehicle-operator-licensing.service.gov.uk',
+  'vehicleenquiry.service.gov.uk',
+  'viewdrivingrecord.service.gov.uk',
+  'view-and-prove-your-rights.homeoffice.gov.uk',
+  'view-immigration-status.service.gov.uk',
+  'visa-address-update.service.gov.uk',
+  'visas-immigration.service.gov.uk',
+  'your-defra-account.defra.gov.uk'
+]
+
+var expected
+var savedUaVars
+var savedGa4Vars
+
+describe('Analytics loading', function () {
+  beforeAll(function () {
+    savedUaVars = window.GOVUK.extendObject(window.GOVUK.analyticsVars)
+    window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+    savedGa4Vars = window.GOVUK.extendObject(window.GOVUK.analyticsGa4.vars)
+  })
+
+  afterAll(function () {
+    window.GOVUK.analyticsGa4.vars = savedGa4Vars
+    window.GOVUK.analyticsVars = savedUaVars
+  })
+
+  describe('for GA4', function () {
+    beforeEach(function () {
+      window.GOVUK.analyticsGa4.vars = null
+      expected = {
+        id: 'GTM-MG7HG5W',
+        auth: undefined,
+        preview: undefined,
+        environment: 'production',
+        gem_version: 'not found'
+      }
+    })
+
+    it('loads GA4 on production domains', function () {
+      var domains = [
+        'www.gov.uk',
+        'www-origin.publishing.service.gov.uk'
+      ]
+
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars.environment).toEqual('production')
+      }
+    })
+
+    it('loads GA4 on staging domains', function () {
+      var domains = [
+        'www.staging.publishing.service.gov.uk',
+        'www.staging.publishing.service.gov.uk',
```

Comment:
Seems to be duplicate entries here again?

Reply:
Yep - meant to be [www-origin.staging.publishing.service.gov.uk](http://www-origin.staging.publishing.service.gov.uk/), good spot